org.eclipse.xtext/src/org/eclipse/xtext/serializer/impl/Serializer.java

```diff
@@ -210,7 +211,28 @@ public ReplaceRegion serializeReplacement(EObject obj, SaveOptions options) {
 			throw new IllegalStateException("Cannot replace an obj that has no associated node");
 		}
 		String text = serialize(obj, options);
-		return new ReplaceRegion(node.getTotalOffset(), node.getTotalLength(), text);
+		int replaceRegionLength = calculateReplaceRegionLength(node, text);
+		return new ReplaceRegion(node.getTotalOffset(), replaceRegionLength, text);
 	}
 
+	private int calculateReplaceRegionLength(ICompositeNode node, String text) {
+		int oldTextLength = node.getTotalLength();
+		int newTextLength = text.length();
+
+		if (newTextLength > oldTextLength) {
+			String remainingText = text.substring(oldTextLength);
+			if (containsOnlyWhiteSpaces(remainingText) && whiteSpaceFollows(node)) {
+				return newTextLength;
+			}
+		}
+		return oldTextLength;
+	}
+
+	private boolean whiteSpaceFollows(ICompositeNode node) {
+		return node.hasNextSibling() && node.getNextSibling() instanceof HiddenLeafNode;
+	}
+
+	private boolean containsOnlyWhiteSpaces(String text) {
```

Comment:
Could be `CharMatcher.whitespac().matchesAllOf(text)`

Reply:
Fixed.